### PR TITLE
Support showing and editing cost center allocation method and basis

### PR DESCRIPTION
**Pull request by jmcameron**
This PR adds the ability to show the allocation method for each cost center and its allocation method.  

This PR partially addresses https://github.com/IMA-WorldHealth/bhima/issues/5899, This is a first step in implementing Issue #5879, and https://github.com/IMA-WorldHealth/bhima/issues/5879.   As it is finalized (or succeeding PRs finish the incomplete parts of this PR), some of those Issues will need to be closed later.

WARNING: This is a partial implementation:  Although you can edit the allocation method, I have not yet implemented editing the allocation basis; it is currently being done by SQL initialization data.   My goal in getting this out now, is so that the cost center step-down report can be updated to use this data before releasing it.

This PR adds the SQL tables 'cost_center_basis' and 'cost_center_basis_value' tables.  That later is not being used yet.  It also adds two members to the fee_center table:  'allocation_method' and 'allocation_method_id'.

NOTE: The goal of the new 'allocation_basis' field is to allow developers to define common allocation basis items (as is done in bhima.sql) but also support hand-created custom basis items.   The registry has a function that determines if it is a translation token and translates it; if not it does not change it.   I'm thinking that as we get this a bit more mature, we can predefine a number of basis items (in Bhima.sql) and reduce the need for creating the same things over and over.   Also one I thought of was "Number of services by center" (and add an optional field that identifies which center it is, perhaps with a units field.  We will not to work though things to decide what the final form should be.

**TESTING**
- Test it using bhima_test
- Show the Fee Center (Cost Center) registry page.  Note the new fields
- Try editing a center and changing its allocation method.

### CODE REVIEW

File: server/models/migrations/next/migrate.sql [hunk 1]
```diff
@@ -41,3 +41,27 @@ CALL add_column_if_missing('general_ledger', 'principal_center_id', 'MEDIUMINT(8
...
 CALL add_constraint_if_missing('general_ledger', 'general_ledger__cost_center_1', 'FOREIGN KEY (`cost_center_id`) REFERENCES `fee_center` (`id`) ON UPDATE CASCADE');
 CALL add_constraint_if_missing('general_ledger', 'general_ledger__cost_center_2', 'FOREIGN KEY (`principal_center_id`) REFERENCES `fee_center` (`id`) ON UPDATE CASCADE');
+
+/**
+author: @jmcameron
+date: 2021-09-01
+description: adds allocation basis data to cost centers
+*/
+DROP TABLE IF EXISTS `cost_center_basis`;
+CREATE TABLE `cost_center_basis` (
```
Comment (mbayopanda) [inline server/models/migrations/next/migrate.sql]: Removing the `DROP TABLE` instruction, and use the `CREATE TABLE IF NOT EXISTS ` will prevent us from the error below when running the migration script.

image

**Review by mbayopanda — APPROVED** [review]
Names of new tables are as we discussed 👍🏽 

New columns are displayed successfully and the update of allocation method for a `fee_center` is working fine.

Adding a new `fee_center` returns a success message but nothing in the grid, if this point is not a blocking point I can approve this PR because it is not the final work (UI for populate the `cost_center_basis` and `cost_center_basis_value`) and others PR depends on this PR

File: server/models/migrations/next/migrate.sql (hunk 1 above)
Comment (jmcameron) [reply server/models/migrations/next/migrate.sql]: I'll push and update.

**Comment by jmcameron** [comment]
> Adding a new `fee_center` returns a success message but nothing in the grid, if this point is not a blocking point I can approve this PR because it is not the final work (UI for populate the `cost_center_basis` and `cost_center_basis_value`) and others PR depends on this PR

I agree. Lets get this out there and fix some of these issues later.

**Pull request merged**